addons/hr_holidays/models/hr_leave_type.py

```diff
@@ -134,8 +156,8 @@ def _search_valid(self, operator, value):
         WHERE
             alloc.employee_id = %s AND
             alloc.active = True AND alloc.state = 'validate' AND
-            (alloc.date_to >= %s OR alloc.date_to IS NULL) AND
-            alloc.date_from <= %s 
+            alloc.date_from <= %s AND
+            (alloc.date_to >= %s OR alloc.date_to IS NULL)
```

Comment:
why change it? unless I'm missing something this is exactly the same code just ordered differently

Reply:
You correct! That's a small refactor (very helpful tho ihmo ) that helps keeping consistency between versions and above all avoid confusion.

Since version 17.0 we consider **date_to** to be the **end** of the event and **date_form** to be the start of the event. So it makes sense to assign them in the same way, and perform "swap" at the time of the query.

https://github.com/odoo/odoo/blob/f80094f43bcc464cc99fba6aa59f64659fe659b6/addons/hr_holidays/models/hr_leave_type.py#L129-L135

Reply:
wasn't this always the case (date_to and date_from being respectively end and start). But reading it again, I was wrong, this seems to be a real change, before we had:

```py
(alloc.date_to >= {date_to} OR alloc.date_to IS NULL) AND
alloc.date_from <= {date_from}
```
which corresponds to checking that the `[{date_to},{date_from}]` is contained in the `[alloc.date_to,alloc.date_from]` range.

and after the change corresponds to:

```py
(alloc.date_to >= {date_from} OR alloc.date_to IS NULL) AND
alloc.date_from <= {date_to}
```
which correspond to `[{date_to},{date_from}]` range having an intersection with `[alloc.date_to,alloc.date_from]` range.

Reply:
In this specific case, it's just reordering two lines, I think it's irrelevant to this fix :) I can stay as it is to avoid unneeded line blame